crates/rpc/rpc-convert/src/transaction.rs

```diff
@@ -55,6 +55,16 @@ pub trait ReceiptConverter<N: NodePrimitives>: Debug + 'static {
         &self,
         receipts: Vec<ConvertReceiptInput<'_, N>>,
     ) -> Result<Vec<Self::RpcReceipt>, Self::Error>;
+
+    /// Converts a set of primitive receipts to RPC representations. It is guaranteed that all
+    /// receipts are from `block`.
+    fn convert_receipts_with_block(
+        &self,
+        receipts: Vec<ConvertReceiptInput<'_, N>>,
+        _block: N::Block,
```

Comment:
I think we can use &N::Block

because most of the type we have this in form of `Arc` and can leave it up to the consumer if they need to clone